shared/main.tf

```diff
@@ -3,3 +3,49 @@ module "ecr" {
 
   namespace = local.app_name
 }
+
+module "iam_groups" {
+  source = "../modules/iam/groups"
+}
+
+module "iam_admin_users" {
+  source = "../modules/iam/users"
+
+  usernames = local.iam_admin_emails
+}
+
+module "iam_developer_users" {
+  source = "../modules/iam/users"
+
+  usernames = local.iam_developer_emails
+}
+
+module "iam_bot_users" {
+  source = "../modules/iam/users"
+
+  usernames = local.iam_bot_emails
+}
+
+module "iam_admin_group_membership" {
+  source = "../modules/iam/group_membership"
+
+  name  = "admin-group-membership"
+  group = module.iam_groups.admin_group
+  users = local.iam_admin_emails
```

Comment:
Based on my experience it is possible that TF plan will fail when we add new user, because not all resources were created yet. For all `group_membership` blocks we need to add depends_on block to solve this potential issue.

```
depends_on = [
    module.iam_groups,
    module.iam_admin_users,
    module.iam_developer_users,
    module.iam_bot_users,
  ]
  ```